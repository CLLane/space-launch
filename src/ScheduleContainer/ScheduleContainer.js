import React from 'react';
import { Cards } from '../Cards/Cards';
import { connect } from 'react-redux';

export const ScheduleContainer = ({ monthLaunch }) => {
  let calendar = monthLaunch.map(launch => <Cards launch={launch} />)

  return (
      <>
      {calendar}
      </>
  )
}

export const mapStateToProps = ({ monthLaunch }) => ({
  monthLaunch
})

export default connect(mapStateToProps)(ScheduleContainer)