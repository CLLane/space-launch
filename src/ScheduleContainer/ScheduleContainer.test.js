import React from 'react';
import { shallow } from 'enzyme'
import { ScheduleContainer, mapStateToProps } from "../ScheduleContainer/ScheduleContainer";

describe('ScheduleContainer', () => {
  let wrapper;
  let mockMonthLaunch = [{name: 'Falcon'}]

  beforeEach(() => {
    wrapper = shallow(
      <ScheduleContainer
        monthLaunch={mockMonthLaunch}
        key={1}
      />
    );
  });
  it('should match a snapshot', () => {
  expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an array with launch objects', () => {
    const mockState = {
      monthLaunch: [{name: 'Falcon'}]
    }
    const expected = {
      monthLaunch: [{ name: 'Falcon' }]
    }
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})