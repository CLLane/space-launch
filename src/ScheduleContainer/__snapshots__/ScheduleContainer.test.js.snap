// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleContainer should match a snapshot 1`] = `
<Fragment>
  <Cards
    launch={
      Object {
        "name": "Falcon",
      }
    }
  />
</Fragment>
`;
